src/main/scala/com/fulcrumgenomics/util/Metric.scala

```diff
@@ -205,6 +223,41 @@ object Metric extends LazyLogging {
   def writer[T <: Metric](writer: Writer)(implicit tt: ru.TypeTag[T]): MetricWriter[T] = new MetricWriter[T](writer)
 }
 
+
+class MetricSorter[Key <: Ordered[Key], T <: Metric](maxObjectsInRam: Int = MetricSorter.MaxInMemory,
+                                                     keyfunc: T => Key,
+                                                     tmpDir: DirPath = Io.tmpDir
+                                                    )(implicit tt: ru.TypeTag[T]) extends Sorter[T, Key](
+  maxObjectsInRam = maxObjectsInRam,
+  codec           = new MetricSorter.MetricSorterCodec[T](),
+  keyfunc         = keyfunc,
+  tmpDir          = tmpDir
+)
+
+object MetricSorter {
+  /** The default maximum # of records to keep and sort in memory. */
+  val MaxInMemory: Int = 1e6.toInt
+
+  class MetricSorterCodec[T <: Metric]()(implicit tt: ru.TypeTag[T])
```

Comment:
could make this into a `def codec[T <: Metric](implicit tt: ru.TypeTag[T]): Sorter.Codec[T] with LazyLogging`